Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -2237,15 +2237,9 @@ void RenderStyle::setContent(RefPtr<StyleImage>&& image, bool add)
 void RenderStyle::setContent(const String& string, bool add)
 {
     auto& data = m_nonInheritedData.access().miscData.access();
-    if (add && data.content) {
-        auto& last = lastContent(*data.content);
-        if (!is<TextContentData>(last))
-            last.setNext(makeUnique<TextContentData>(string));
-        else {
-            auto& textContent = downcast<TextContentData>(last);
-            textContent.setText(textContent.text() + string);
-        }
-    } else {
+    if (add && data.content)
+        lastContent(*data.content).setNext(makeUnique<TextContentData>(string));
```

Comment:
Is there no room for an optimization where we concatenate? I understand that the old code didn't separate appropriately.

Or maybe it's not an effective optimization or impossible to do correctly.

Reply:
We could optimize this by storing concatenated data as a separate list, but I don't think it's worth doing because `content` data is typically very short.